File: EXAM/gongora/u3/GongoraStrategy/SortApp/java/Strategy/src/main/java/ec/edu/espe/strategy/model/BubbleSort.java
package ec.edu.espe.strategy.model;

/**
 *
 * @author Lucas Gongora
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int[] data) {
        int n = data.length;
        boolean swapped;
        do {
            swapped = false;
            for (int i = 0; i < n - 1; i++) {
                if (data[i] > data[i + 1]) {
                    int temp = data[i];
                    data[i] = data[i + 1];
                    data[i + 1] = temp;
                    swapped = true;
                }
            }
            n--;
        } while (swapped);
        return data;
    }
}

File: EXAM/camacho/u3/CamachoStrategy/pooCrudjs/public/model/BubbleSort.js
import  SortingStrategy from "./SortingStrategy.js"

export default class BubbleSort extends SortingStrategy{
    sort(data) {
        let n = data.length;
        let swapped;

        do {
            swapped = false;
            for (let i = 0; i < n - 1; i++) {
                if (data[i] > data[i + 1]) {
                    
                    [data[i], data[i + 1]] = [data[i + 1], data[i]];
                    swapped = true;
                }
            }
            n--;
        } while (swapped);

        return data;
    }
}

